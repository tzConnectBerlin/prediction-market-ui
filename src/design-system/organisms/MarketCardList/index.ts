export { MarketCardList, MarketCardList as default } from './MarketCardList';
