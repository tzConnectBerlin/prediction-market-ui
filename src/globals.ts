import { WalletProviderProps } from '@tezos-contrib/react-wallet-provider';

const network = process.env.REACT_APP_NETWORK_TYPE || 'GRANADANET';

export const APP_NAME = process.env.REACT_APP_APP_NAME || 'Prediction Market';
export const NETWORK: WalletProviderProps['network'] = network as WalletProviderProps['network'];
export const MARKET_ADDRESS = process.env.REACT_APP_MARKET_CONTRACT || '';
export const FA12_CONTRACT = process.env.REACT_APP_FA12_CONTRACT || '';
export const RPC_URL = process.env.REACT_APP_RPC_URL || 'https://granadanet.smartpy.io';
export const RPC_PORT = process.env.REACT_APP_RPC_PORT || 443;
export const IPFS_POST_API = process.env.REACT_APP_IPFS_POST_API;
export const IPFS_GET_API = process.env.REACT_APP_IPFS_GET_API;
export const GRAPHQL_API = process.env.REACT_APP_GRAPHQL_API || '';
export const DATETIME_FORMAT = {
  LONG_FORMAT: 'PPPPpppp',
  MEDIUM_FORMAT: 'do MMM yyyy HH:mm',
  SHORT_FORMAT: 'do MMM yyyy',
  INPUT_FORMAT: 'dd/MM/yyyy',
};
export const SENTRY_DSN = process.env.REACT_APP_SENTRY_DSN;
export const CURRENCY_SYMBOL = process.env.REACT_APP_FA12_SYMBOL || 'PMM';
export const ENABLE_MARKET_CREATION =
  process.env.REACT_APP_ENABLE_MARKET_CREATION === 'true' || false;
export const WERT_PARTNER_ID = process.env.REACT_APP_WERT_PARTNER_ID;
export const TORUS_ENABLED = process.env.REACT_APP_TORUS_ENABLED === 'true';
export const AUTH_URL = process.env.REACT_APP_AUTH_URL;
export const AUTH_SCOPE = process.env.REACT_APP_AUTH_SCOPE ?? 'publicData';
export const AUTH_REDIRECT_URL = process.env.REACT_APP_AUTH_REDIRECT_URL;
export const AUTH_CLIENT_ID = process.env.REACT_APP_AUTH_CLIENT_ID;
export const TORUS_NETWORK = process.env.REACT_APP_TORUS_NETWORK ?? 'testnet';
export const TORUS_PROVIDER = process.env.REACT_APP_TORUS_PROVIDER;
export const MARKET_CRREATION = process.env.REACT_APP_MARKET_CREATOR;

export const SLIPPAGE = 5;
export const DEADLINE = 30;
