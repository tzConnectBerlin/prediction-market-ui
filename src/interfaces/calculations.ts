export interface ClosePositionReturn {
  aToSwap: number;
  aLeft: number;
  bReceived: number;
  aToSwapWithSlippage: number;
}

export interface ClosePositionBothReturn
  extends Omit<ClosePositionReturn, 'bReceived' | 'aToSwapWithSlippage'> {
  bHeld: number;
}

export interface BuyPosition {
  price: number;
  swap: number;
  quantity: number;
}
