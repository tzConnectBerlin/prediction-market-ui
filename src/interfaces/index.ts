export * from './wallet';
export * from './router';
