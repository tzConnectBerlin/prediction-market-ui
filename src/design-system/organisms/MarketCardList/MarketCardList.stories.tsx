import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { MarketCardData, MarketStateType, TokenType } from '../../../interfaces/market';
import { MarketCardList, MarketCardListProps } from './MarketCardList';

export default {
  title: 'Organism/MarketCardList',
  component: MarketCardList,
} as Meta;

const Template: Story<MarketCardListProps> = (args) => <MarketCardList {...args} />;
const dataList: MarketCardData[] = [
  {
    marketId: '1',
    block: 1,
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06T07:19:00.000Z',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    bakedAt: '2021-06-08T15:36:11+02:00',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',
        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
  {
    marketId: '1',
    block: 1,
    bakedAt: '',
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06T07:19:00.000Z',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',

        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
  {
    marketId: '1',
    block: 1,
    bakedAt: '',
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06T07:19:00.000Z',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',

        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
  {
    marketId: '1',
    block: 1,
    bakedAt: '',
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06T07:19:00.000Z',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',

        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
  {
    marketId: '1',
    block: 1,
    bakedAt: '',
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06T07:19:00.000Z',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',
        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
];
export const Default = Template.bind({});
Default.args = {
  cardList: dataList,
};
