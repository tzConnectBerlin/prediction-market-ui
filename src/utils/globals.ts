import { NetworkType } from '../interfaces';

export const APP_NAME = process.env.REACT_APP_APP_NAME || 'Prediction Market';
export const NETWORK = (process.env.REACT_APP_NETWORK_TYPE || 'edonet') as NetworkType;
export const MARKET_ADDRESS = process.env.REACT_APP_MARKET_CONTRACT;
export const FA12_CONTRACT = process.env.REACT_APP_FA12_CONTRACT;
export const RPC_URL = process.env.REACT_APP_RPC_URL || 'https://edonet.smartpy.com';
export const RPC_PORT = process.env.REACT_APP_RPC_PORT || 443;
export const IPFS_PORT = process.env.REACT_APP_IPFS_PORT;
export const IPFS_API = process.env.REACT_APP_IPFS_API;
export const BCD_BASE_API = process.env.REACT_APP_BASE_BCD_API || 'https://api.better-call.dev/v1/';
export const ENABLE_SAME_MARKETS = process.env.REACT_APP_ENABLE_SAME_MARKETS === 'true';
export const ENABLE_SIMILAR_MARKETS = process.env.REACT_APP_ENABLE_SIMILAR_MARKETS === 'true';
export const TEZOS_MDW_API =
  process.env.REACT_APP_TEZOS_MDW || 'https://pmcache.newby.org/markets.json';
export const DATETIME_FORMAT = {
  LONG_FORMAT: 'do MMM yyyy HH:mm:ss',
  MEDIUM_FORMAT: 'do MMM yyyy HH:mm',
  SHORT_FORMAT: 'do MMM yyyy',
  INPUT_FORMAT: 'dd/MM/yyyy',
};
export const SENTRY_DSN = process.env.REACT_APP_SENTRY_DSN;
