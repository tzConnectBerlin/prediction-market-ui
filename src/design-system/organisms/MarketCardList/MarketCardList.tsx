import * as React from 'react';
import { Grid, useTheme } from '@material-ui/core';
import { motion } from 'framer-motion';
import { useHistory } from 'react-router-dom';
import styled from '@emotion/styled';
import { useTranslation } from 'react-i18next';
import { format } from 'date-fns-tz';
import { MarketCard } from '../MarketCard';
import { MarketCardData, MarketCardToken, TokenType } from '../../../interfaces';
import { roundToTwo } from '../../../utils/math';
import { SkeletonCard } from '../SkeletonCard';
import { questionToURL } from '../../../utils/misc';
import { CURRENCY_SYMBOL } from '../../../globals';

const StyledGrid = styled(Grid)`
  display: flex;
`;

const StyledMotionDiv = styled(motion.div)`
  display: flex;
`;

export interface MarketCardListProps {
  pending?: number;
  cardList: MarketCardData[];
  timestampFormat?: string;
}

const container = {
  hidden: { opacity: 0 },
  show: {
    opacity: 1,
    transition: {
      staggerChildren: 0.2,
      delayChildren: 0.2,
    },
  },
};

const item = {
  hidden: { opacity: 0 },
  show: { opacity: 1 },
};

export const MarketCardList: React.FC<MarketCardListProps> = ({ cardList, pending = 0 }) => {
  const { t } = useTranslation(['common']);
  const history = useHistory();
  const theme = useTheme();

  const getMarketList = () => {
    return cardList.map((card, index) => {
      const cardLink = questionToURL(card.question);
      const yes = Number.isNaN(card.yesPrice) ? '--' : card.yesPrice;
      const no = Number.isNaN(card.yesPrice) ? '--' : roundToTwo(1 - card.yesPrice);
      const stats = [];
      const phase =
        t(card.state).toLowerCase() === 'auction'
          ? t('auctionPhase')
          : card?.winningPrediction
          ? t('resolved')
          : t('marketPhase');
      if (card?.winningPrediction) {
        stats.push({
          type: t('resolution'),
          value: card.winningPrediction.toUpperCase(),
        });
        stats.push({
          type: t('resolvedOn'),
          value: format(new Date(card.bakedAt), 'PP'),
        });
      } else {
        if (card.weekly) {
          stats.push({
            type: t('weekly'),
            value: `+${card.weekly.change}%`,
            tokenType: card.weekly.tokenType,
          });
        } else if (!card.weekly && !card.winningPrediction) {
          stats.push({
            type: t('weekly'),
            value: '---',
          });
        }
        stats.push({
          type: t('liquidity'),
          value: card.liquidity ? `${card.liquidity} ${CURRENCY_SYMBOL}` : '--',
        });
      }

      let backgroundColor;
      let fontColor;
      if (t(card.state).toLowerCase() === 'auction') {
        backgroundColor = theme.palette.secondary.dark;
        fontColor = theme.palette.text.primary;
      }
      if (card.winningPrediction) {
        // eslint-disable-next-line prefer-destructuring
        backgroundColor = theme.palette.grey[400];
        fontColor = theme.palette.text.primary;
      }

      const tokenList: MarketCardToken[] = [];

      if (typeof yes !== 'string') {
        tokenList.push({
          type: TokenType.yes,
          value: roundToTwo(yes * 100),
        });
      }

      if (typeof no !== 'string') {
        tokenList.push({
          type: TokenType.no,
          value: roundToTwo(no * 100),
        });
      }

      return (
        <StyledMotionDiv variants={item} key={`${card?.ipfsHash}-${index}`}>
          <StyledGrid item>
            <MarketCard
              title={card.question}
              hash={card.ipfsHash}
              cardState={phase}
              onClick={() => history.push(`/market/${card.marketId}/${cardLink}`)}
              cardStateProps={{
                backgroundColor,
                fontColor,
              }}
              iconURL={card.iconURL}
              tokenList={tokenList}
              statisticList={stats}
            />
          </StyledGrid>
        </StyledMotionDiv>
      );
    });
  };

  return (
    <motion.div variants={container} initial="hidden" animate="show">
      <Grid justifyContent="flex-start" container role="grid">
        {pending > 0 &&
          new Array(pending).fill('').map((_, index) => <SkeletonCard key={`skeleton-${index}`} />)}
        {getMarketList()}
      </Grid>
    </motion.div>
  );
};
