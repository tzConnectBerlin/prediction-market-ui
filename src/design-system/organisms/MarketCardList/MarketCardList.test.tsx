import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { MarketCardData, MarketStateType, TokenType } from '../../../interfaces';
import { MarketCardList } from './MarketCardList';
import { DATETIME_FORMAT } from '../../../globals';

const dataList: MarketCardData[] = [
  {
    marketId: '1',
    bakedAt: '',
    block: 1,
    adjudicator: 'tk1111',
    description: 'This is sample description',
    question: 'This is a question',
    auctionEndDate: '2021-04-06',
    yesPrice: 0.5,
    state: MarketStateType.auctionRunning,
    ipfsHash: '',
    iconURL: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
    ticker: 'ticker',
    tokens: [
      {
        type: TokenType.yes,
        value: 14,
      },
      {
        type: TokenType.no,
        value: 5,
      },
    ],
    statistics: [
      {
        type: 'WEEKLY',
        value: '95',
        changes: 'up',
        tokenType: TokenType.yes,
      },
      {
        type: 'VOLUME',
        value: '5',
      },
    ],
  },
];
describe('Snapshot testing MarketCardList Component', () => {
  it('renders correctly', () => {
    const List = renderer
      .create(
        <MemoryRouter>
          <MarketCardList cardList={dataList} timestampFormat={DATETIME_FORMAT.SHORT_FORMAT} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(List).toMatchSnapshot();
  });
});

describe('Element testing MarketCardList Component', () => {
  it('render correctly 1 item', () => {
    const { getAllByText } = render(
      <MemoryRouter>
        <MarketCardList cardList={dataList} />
      </MemoryRouter>,
    );

    expect(getAllByText('liquidity').length).toBe(1);
  });
});
