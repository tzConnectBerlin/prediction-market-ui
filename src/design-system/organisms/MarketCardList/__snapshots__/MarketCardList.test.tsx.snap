// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketCardList Component renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:599.95px) {
  .emotion-1 {
    width: 100%;
  }
}

@media (min-width:600px) and (max-width:899.95px) {
  .emotion-1 {
    width: 50%;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    width: 33.33%;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  margin: 0;
  color: #1976d2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  margin: 1em;
  width: 100%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-7.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-7.xs {
  width: 24px;
  height: 24px;
}

.emotion-7.sm {
  width: 32px;
  height: 32px;
}

.emotion-7.md {
  width: 40px;
  height: 40px;
}

.emotion-7.lg {
  width: 48px;
  height: 48px;
}

.emotion-7.xl {
  width: 64px;
  height: 64px;
}

.emotion-7.xxl {
  width: 77px;
  height: 77px;
}

.emotion-7.max {
  width: 144px;
  height: 144px;
}

.emotion-8 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: block;
}

.emotion-11 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
}

.emotion-11>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-11>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-13 {
  border-radius: 0.2em;
  padding: 0.3em 0.6em;
  display: inline-block;
  background-color: #7b1fa2;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-13.label--icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-13.label--icon>* {
  margin-right: 0.3em;
}

.emotion-13.label--icon>svg {
  width: 0.8em;
  height: 0.8em;
}

.emotion-14 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-14.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-15 {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  padding: 0.3em 0;
}

.emotion-16 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  font-weight: 700;
}

.emotion-16.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-17 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  padding-top: 0;
  margin-top: auto;
}

.emotion-17>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-17>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-18 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-flex-basis: calc(100% + 24px);
  -ms-flex-preferred-size: calc(100% + 24px);
  flex-basis: calc(100% + 24px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 24px);
}

.emotion-18>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-18>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

.emotion-20 {
  color: rgba(29, 34, 39, 0.65);
  padding: 0.2em;
}

.emotion-20.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-20.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-21 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-21.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-22 {
  color: rgba(42, 184, 0, 1);
  padding: 0.2em;
}

.emotion-22.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-22.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-23 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-23.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-27 {
  color: rgba(255, 57, 57, 1);
  padding: 0.2em;
}

.emotion-27.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-27.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-33 {
  padding: 0.2em;
}

.emotion-33.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-33.hasIcon>* {
  margin-left: 0.2em;
}

<div
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-0"
    role="grid"
  >
    <div
      className="emotion-1"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone emotion-3"
          href="/market/1/this-is-a-question"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-4"
          >
            <div
              className="MuiCardHeader-root emotion-5"
            >
              <div
                className="MuiCardHeader-avatar emotion-6"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-7"
                >
                  <img
                    alt=""
                    className="MuiAvatar-img emotion-8"
                    src="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiCardHeader-content emotion-9"
              >
                <span
                  className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-10"
                >
                  <div
                    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-11"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item emotion-12"
                    >
                      <span
                        className=" emotion-13"
                      >
                        <div
                          className="MuiTypography-root MuiTypography-h5  emotion-14"
                          lines={3}
                        >
                          auctionPhase
                        </div>
                      </span>
                    </div>
                  </div>
                  <div
                    className="emotion-15"
                  >
                    <h2
                      className="MuiTypography-root MuiTypography-h2  emotion-16"
                      lines={3}
                    >
                      This is a question
                    </h2>
                  </div>
                </span>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-17"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-18"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-19"
                >
                  <div
                    className="emotion-20"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h4  emotion-21"
                      lines={3}
                    >
                      Yes
                    </h3>
                  </div>
                  <div
                    className="emotion-22"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h3  emotion-23"
                      lines={3}
                    >
                      50
                      %
                    </h3>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-19"
                >
                  <div
                    className="emotion-20"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h4  emotion-21"
                      lines={3}
                    >
                      No
                    </h3>
                  </div>
                  <div
                    className="emotion-27"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h3  emotion-23"
                      lines={3}
                    >
                      50
                      %
                    </h3>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-18"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-19"
                >
                  <div
                    className=" emotion-20"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h4  emotion-21"
                      lines={3}
                    >
                      weekly
                    </h3>
                  </div>
                  <div
                    className="emotion-33"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h3  emotion-23"
                      lines={3}
                    >
                       
                      ---
                    </h3>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-19"
                >
                  <div
                    className=" emotion-20"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h4  emotion-21"
                      lines={3}
                    >
                      liquidity
                    </h3>
                  </div>
                  <div
                    className="emotion-33"
                  >
                    <h3
                      className="MuiTypography-root MuiTypography-h3  emotion-23"
                      lines={3}
                    >
                       
                      --
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
